src/main/kotlin/no/nav/familie/ef/iverksett/økonomi/simulering/kontroll/SimuleringskontrollService.kt

```diff
@@ -37,31 +50,95 @@ class SimuleringskontrollService(
 
         val behandlingId = iverksett.behandling.behandlingId
         try {
-            simulerMedNyUtbetalingsgeneratorOgSjekkDif(simulering, beriketSimuleringsresultat)
+            simulerMedGammelUtbetalingsgeneratorOgSjekkDiff(simulering, beriketSimuleringsresultat)
         } catch (e: Exception) {
             logger.warn("Feilet kontroll av behandling=$behandlingId")
         }
     }
 
-    private fun simulerMedNyUtbetalingsgeneratorOgSjekkDif(
+    private fun simulerMedGammelUtbetalingsgeneratorOgSjekkDiff(
         simulering: Simulering,
         beriketSimuleringsresultat: BeriketSimuleringsresultat,
     ) {
         val behandlingId = simulering.nyTilkjentYtelseMedMetaData.behandlingId
-        val simuleringNyUtbetalingsgenerator =
-            simuleringService.hentBeriketSimulering(simulering, brukNyUtbetalingsgenerator = true)
+        val simuleringsresultatKontroll = hentKontrollsimulering(simulering)
         val tidligerePerioder = beriketSimuleringsresultat.oppsummering.perioder.sortedBy { it.fom }
-        val nyePerioder = simuleringNyUtbetalingsgenerator.oppsummering.perioder.sortedBy { it.fom }
+        val nyePerioder = simuleringsresultatKontroll.oppsummering.perioder.sortedBy { it.fom }
 
         val harDiff = harDiff(behandlingId, tidligerePerioder, nyePerioder)
         if (harDiff) {
-            val resultat = SimuleringskontrollResultat(simuleringNyUtbetalingsgenerator)
+            val resultat = SimuleringskontrollResultat(simuleringsresultatKontroll)
             val input = SimuleringskontrollInput(simulering, beriketSimuleringsresultat)
             simuleringskontrollRepository.insert(Simuleringskontroll(behandlingId, input, resultat))
         }
         logger.info("behandling=$behandlingId - kontroll av ny utbetalingsgenerator utført - harDiff=$harDiff")
     }
 
+    private fun hentKontrollsimulering(simulering: Simulering): BeriketSimuleringsresultat {
+        val detaljertSimuleringResultat = hentKontrollResultat(simulering)
+        val simuleringsresultatDto = lagSimuleringsoppsummering(detaljertSimuleringResultat, LocalDate.now())
+
+        return BeriketSimuleringsresultat(
+            detaljer = detaljertSimuleringResultat,
+            oppsummering = simuleringsresultatDto,
+        )
+    }
+
+    private fun hentKontrollResultat(
+        simulering: Simulering,
+    ): DetaljertSimuleringResultat {
+        if (featureToggleService.isEnabled("familie.ef.iverksett.stopp-iverksetting")) {
+            error("Kan ikke sende inn simmulere")
```

Comment:
Kanskje nevne i feilmeldingen at det ikke kan simuleres fordi featuretoggle er skrudd på

Reply:
`Iverksetting er skrudd av - kan ikke simulere nå`

Går for en slik variant.